### PR TITLE
🌱 [wip] kubetest: Allow kubetest provider to be overridable

### PR DESCRIPTION
**Pull request by randomvariable**
**What this PR does / why we need it**:
kubetest previously defaulted to the "skeleton" provider, which
blocks cloud provider specific tests for SIG Networking and SIG Storage
to be run, e.g. Service Type=LoadBalancer or in-tree PVC tests.

Signed-off-by: Naadir Jeewa <jeewan@vmware.com>

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by randomvariable** [comment]
Also need to allow providers to set env vars for the docker container so kubetest can contact the cloud provider.

File: test/framework/kubetest/run.go [hunk 1]
```diff
@@ -143,6 +144,9 @@ func Run(ctx context.Context, input RunInput) error {
 		tmpKubeConfigPath: "/tmp/kubeconfig",
 		reportDir:         "/output",
 	}
+	for k, v := range input.ExtraMounts {
+		volumeMounts[k] = v
+	}
 	user, err := user.Current()
 	if err != nil {
 		return errors.Wrap(err, "unable to determine current user")
```
Comment (sbueringer) [inline test/framework/kubetest/run.go:152]: @randomvariable Does it make a difference that before it was part of `e2eArgs` and now of `config` (and there's another `--` in between) (line 171/172)

Comment (randomvariable) [reply test/framework/kubetest/run.go:152]: seemingly not.

Comment (sbueringer) [reply test/framework/kubetest/run.go:152]: Is that based on tests in CAPA or CAPI? I would run our test which uses this part, but I have to fix this test first.

Comment (randomvariable) [reply test/framework/kubetest/run.go:152]: Fiddling around in CAPA at the moment.

**Comment by fabriziopandini** [comment]
@sbueringer @yastij does it makes sense to revive this effort? Is still a need on the provider's side?

**Comment by sbueringer** [comment]
> @sbueringer @yastij does it makes sense to revive this effort? Is still a need on the provider's side?

It seems useful to me, but only if there is interest by providers for running those additional tests.

**Comment by fabriziopandini** [comment]
/lifecycle frozen
To get this moving it is required to figure it out how to integrate with test infra, something similar to https://github.com/kubernetes/kops/tree/master/tests/e2e/kubetest2-kops but once we have all the pieces in place so this will allow external projects to use CAPI as their test system (and thus finally drop /k/k/cluster

**Pull request closed without merging**